snapcraft/cli/__init__.py

```diff
@@ -89,6 +89,7 @@ def get_command(self, ctx, cmd_name):
 @click.option('--debug', '-d', is_flag=True)
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
+    ctx.obj['debug'] = debug
     if debug:
```

Comment:
it is needed for the case `snapcraft --debug cleanbuild` the sole reason for this. If `--debug` comes before the command it is not passed on to the sucommand